tests/startup_capture_test.go

```diff
@@ -0,0 +1,270 @@
+		if err = test.tearDownFunction(context); err != nil {
+			helper.ExecCmds(t, test.tearDownCmds...)
+			context.getSystemState(t)
+			t.Fatalf("Fail to tear test down: %s", err.Error())
+		}
+	}
+
+	helper.ExecCmds(t, test.tearDownCmds...)
+
+	if test.mode == Replay {
+		for i, check := range test.checks {
+			checkContext := test.checkContexts[i]
+			t.Logf("Replaying test with time %s (Unix: %d), startTime %s (Unix: %d)", checkContext.time, checkContext.time.Unix(), checkContext.startTime, checkContext.startTime.Unix())
+			err = common.Retry(func() error {
+				return check(checkContext)
+			}, retries, time.Second)
+
+			if err != nil {
+				t.Errorf("Failed to replay test: %s, graph: %s, flows: %s", err.Error(), checkContext.getWholeGraph(t, checkContext.time), checkContext.getAllFlows(t, checkContext.time))
+			}
+		}
+	}
+}
+
+func TestStartupCaptureNew(t *testing.T) {
+	test := &Test{
+		setupCmds: []helper.Cmd{
+			{"ovs-vsctl add-br br-startup", true},
+
+			{"ovs-vsctl add-port br-startup fh-intf1 -- set interface fh-intf1 type=internal", true},
+			{"ip netns add fh-vm1", true},
+			{"ip link set fh-intf1 netns fh-vm1", true},
+			{"ip netns exec fh-vm1 ip address add 169.254.33.33/24 dev fh-intf1", true},
+			{"ip netns exec fh-vm1 ip link set fh-intf1 up", true},
+
+			{"ovs-vsctl add-port br-startup fh-intf2 -- set interface fh-intf2 type=internal", true},
+			{"ip netns add fh-vm2", true},
+			{"ip link set fh-intf2 netns fh-vm2", true},
+			{"ip netns exec fh-vm2 ip address add 169.254.33.34/24 dev fh-intf2", true},
+			{"ip netns exec fh-vm2 ip link set fh-intf2 up", true},
+		},
+
+		settleFunction: func(c *TestContext) error {
+			// check that src and dst interfaces are in the right place before doing the ping
+			gremlin := `G.V().Has("Name", "fh-vm1").Out().Has("Name", "fh-intf1")`
+			nodes, err := c.gh.GetNodes(gremlin)
+			if err != nil || len(nodes) == 0 {
+				return errors.New("fh-intf1 not found in the expected namespace")
+			}
+
+			gremlin = `G.V().Has("Name", "fh-vm2").Out().Has("Name", "fh-intf2")`
+			nodes, err = c.gh.GetNodes(gremlin)
+			if err != nil || len(nodes) == 0 {
+				return errors.New("fh-intf2 not found in the expected namespace")
+			}
+
+			return nil
+		},
+
+		setupFunction: func(c *TestContext) error {
+			helper.ExecCmds(t, helper.Cmd{Cmd: "ip netns exec fh-vm1 ping -c 1 -s 1024 169.254.33.34", Check: false})
+			return nil
+		},
+
+		tearDownCmds: []helper.Cmd{
+			{"ip netns del fh-vm1", true},
+			{"ip netns del fh-vm2", true},
+			{"ovs-vsctl del-br br-startup", true},
+		},
+
+		// since the agent update ticker is about 10 sec according to the configuration
+		// we should wait 11 sec to have the first update and the MetricRange filled
+		checks: []CheckFunction{func(c *CheckContext) error {
+			prefix := "g"
+			if !c.time.IsZero() {
+				prefix += fmt.Sprintf(".Context(%d)", common.UnixMillis(c.time))
+			}
+			prefix += `.V().Has("Name", "br-startup", "Type", "ovsbridge")`
```

Comment:
I would rather test that this node has Capture.ID field and that should be enough for this test